src/diffpy/utils/diffraction_objects.py

```diff
@@ -276,7 +289,7 @@ def get_array_index(self, value, xtype=None):
         """
 
         if xtype is None:
-            xtype = self.input_xtype
+            xtype = self._xtype
```

Comment:
If we do not want the user from directly modifying `xtype`, like `_all_arrays`, `_xtype` is used internally.

Reply:
I like how you changed input_xtype to xtype! I do think people can directly modify xtype though (@sbillinge please confirm). For example I might want to plot my diffraction object on tth first, and then change its xtype to q so I can visualize it in qspace.

Reply:
@yucongalicechen that is not how it works.  the attribute was called `input_xtype` because we wanted to keep a record of the xtype input by the user.  So I don't want to change its name.  When xtype is used for plotting any xtype can be given, for example `diffobj.on_xtype('q')` would satisfy @yucongalicechen 's UC even if the data were input on a `tth` grid.

Since we only use it internally, we could change the attribute name to `_input_xtype` to make it private.  I think it should only be set by `insert_data` and not re-settable.

Reply:
@sbillinge  - done. `input_type` is used internally and is it only gettable.